platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/network/NetworkPeerIdentifierTest.java

```diff
@@ -0,0 +1,134 @@
+import com.swirlds.platform.crypto.CryptoStatic;
+import com.swirlds.platform.crypto.KeyCertPurpose;
+import com.swirlds.platform.crypto.KeyGeneratingException;
+import com.swirlds.platform.crypto.KeyLoadingException;
+import com.swirlds.platform.crypto.PublicStores;
+import java.net.URISyntaxException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.SecureRandom;
+import java.security.cert.Certificate;
+import java.security.cert.PKIXParameters;
+import java.security.cert.TrustAnchor;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+class NetworkPeerIdentifierTest {
+    /**
+     * Tests that given a list of valid Swirlds production certificates (like the type used in mainnet),
+     * {@link NetworkPeerIdentifier#identifyTlsPeer(Certificate[], List)} is able to successfully identify a matching
+     * peer.
+     */
+    @Test
+    void testExtractPeerInfoWorksForMainnet()
+            throws URISyntaxException, KeyLoadingException, KeyStoreException, InvalidAlgorithmParameterException {
+
+        // sample node names from mainnet
+        final List<String> names =
+                List.of("node1", "node2", "node3", "node4", "node5", "node6", "node7", "node8", "node9", "node10");
+        // sample pfx file grabbed from mainnet
+        final KeyStore publicKeys = loadKeys(
+                ResourceLoader.getFile("preGeneratedKeysAndCerts/").resolve("publicMainnet.pfx"),
+                "password".toCharArray());
+        final PublicStores publicStores = PublicStores.fromAllPublic(publicKeys, names);
+
+        final List<PeerInfo> peerInfoList = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            final String name = names.get(i);
+            final NodeId node = new NodeId(i);
+            final PeerInfo peer = new PeerInfo(
+                    node,
+                    names.get(i),
+                    "127.0.0.1",
+                    Objects.requireNonNull(publicStores.getCertificate(KeyCertPurpose.SIGNING, name)));
+            peerInfoList.add(peer);
+        }
+
+        final PKIXParameters params = new PKIXParameters(publicStores.agrTrustStore());
+        final Set<TrustAnchor> trustAnchors = params.getTrustAnchors();
+
+        final Certificate[] certificates =
+                trustAnchors.stream().map(TrustAnchor::getTrustedCert).toArray(Certificate[]::new);
+        final PeerInfo matchedPeer = new NetworkPeerIdentifier().identifyTlsPeer(certificates, peerInfoList);
+        Assertions.assertNotNull(matchedPeer);
```

Comment:
If the method under test simply returned a random or incorrect `PeerInfo` from the list provided, this test would pass. Best to add more assertions to verify that the correct peer was returned.

Reply:
I agree it's generally correct to assert specific properties of an object in tests but we can't do that here as:
- the certificates loaded from the key file are unordered
- we return the first match, not the only match.

Reply:
Got it. Changed to actually match each node, put them into a set, and assert the final set has the exact same elements as the original list of peers.

Reply:
and also added one more test to identify a specific node.